python/cudf/cudf/core/column/datetime.py

```diff
@@ -82,10 +86,29 @@ def from_numpy(cls, array):
                 ("Cannot infer datetime dtype " + "from np.array dtype `%s`")
                 % (array.dtype)
             )
+
+        pa_arr = pa.array(array)
+        buffers = pa_arr.buffers()
+        if buffers[0]:
+            mask_dev_array = utils.make_mask(len(pa_arr))
+            arrow_dev_array = rmm.to_device(
+                np.array(pa_arr.buffers()[0]).view("int8")
```

Comment:
This returns by copy, you should use the buffer by reference instead:
```suggestion
                np.frombuffer(pa_arr.buffers()[0], dtype="int8")
```